javalib/src/main/scala/java/util/concurrent/locks/ReentrantLock.scala

```diff
@@ -0,0 +1,48 @@
+package java.util.concurrent.locks
+
+import java.io.Serializable
+import java.lang.Thread
+import java.util.concurrent.TimeUnit
+
+class ReentrantLock(fair: Boolean) extends Lock with Serializable {
+
+  private var locked = 0
+
+  def this() = this(false)
+
+  def lock(): Unit = locked += 1
+
+  def lockInterruptibly(): Unit =
+    if (Thread.currentThread().isInterrupted()) throw new InterruptedException()
+    else locked += 1
+
+  def tryLock(): Boolean = {
+    locked += 1
+    true
+  }
+
+  def tryLock(time: Long, unit: TimeUnit): Boolean = {
+    if (Thread.currentThread().isInterrupted()) throw new InterruptedException()
+    tryLock()
+  }
+
+  def unlock(): Unit = {
+    if (locked <= 0) throw new IllegalMonitorStateException()
+    else locked -= 1
+  }
+
+  //Not implemented:
+  //def newCondition(): Condition
+
+  def getHoldCount(): Int = locked
+
+  def isHeldByCurrentThread(): Boolean = isLocked()
+
+  def isLocked(): Boolean = locked > 0
+
+  final def isFair(): Boolean = fair
+
+  protected def getOwner(): Thread =
+    if (isLocked) Thread.currentThread()
+    else null
+}
```

Comment:
I don't understand what you mean by that?

Oh, btw, you can't avoid `toString()`. That one is necessary since it's an override from a superclass, so it will link anyway if it's not implemented. But `toString()` is easy, it's like `super.toString() + "[" + (if (isLocked) "Locked by " + Thread.currentThread.name else "Unlocked") + "]"` (but better written)